### PR TITLE
ObservationWithRubrics report

### PR DESCRIPTION
**Pull request by praveenKDass**

### CODE REVIEW

File: controllers/v1/reportsController.js [hunk 1]
```diff
@@ -29,7 +29,44 @@ module.exports = class Reports {
   static get name() {
     return 'submissions';
   }
-
+  
+ async fetch  (req, res) {
```
Comment (VISHNUDAS-tunerlabs) [inline controllers/v1/reportsController.js:33]: @praveenKDass why there i two function with same name?

File: controllers/v1/reportsController.js [hunk 2]
```diff
@@ -29,7 +29,44 @@ module.exports = class Reports {
...
+
+  try {
```
Comment (VISHNUDAS-tunerlabs) [inline controllers/v1/reportsController.js:35]: @praveenKDass We have to finalise @borkarsaish65 changes and incorporate them correctly here after testing

**Review by VISHNUDAS-tunerlabs — CHANGES_REQUESTED** [review]

File: controllers/v1/reportsController.js (hunk 1 above)
Comment (praveenKDass) [reply controllers/v1/reportsController.js]: removed @VISHNUDAS-tunerlabs

File: helper/pdfGeneration.js [hunk 3]
```diff
@@ -1076,4 +1076,326 @@ const uploadPdfToCloud = async function(fileName, folderPath) {
...
+              dir + "/header.html",
+              headerHtml,
+              function (errWr, dataWr) {
+                if (errWr) {
+                  throw errWr;
+                } else {
+                  var obj = {
+                    path: formDataAssessment,
+                  };
+                  console.log("data:", obj.path[0].options.filename);
```
Comment (VISHNUDAS-tunerlabs) [inline helper/pdfGeneration.js:1152]: @praveenKDass remove unwanted console from every where

File: helper/pdfGeneration.js [hunk 4]
```diff
@@ -1076,4 +1076,326 @@ const uploadPdfToCloud = async function(fileName, folderPath) {
...
+    submissionId=""
+  ) {
+    return new Promise(async function (resolve, reject) {
+      let currentTempFolder =
+        "tmp/" +
+        uuidv4() +
+        "--" +
+        Math.floor(Math.random() * (10000 - 10 + 1) + 10);
+  
+      let imgPath = __dirname + "/../" + currentTempFolder;
```
Comment (VISHNUDAS-tunerlabs) [inline helper/pdfGeneration.js:1102]: @praveenKDass this type of path assign can cause error in server
better to do 
let imgPath = path.resolve(__dirname, '../../', currentTempFolder)
and test. make the change every where. even in @borkarsaish65 files also you check it

File: module/programs/helper.js [hunk 5]
```diff
@@ -6,15 +6,11 @@
  */
 
 // Dependencies
-const entitiesHelper = require(MODULES_BASE_PATH + '/entities/helper');
 const timeZoneDifference = process.env.TIMEZONE_DIFFRENECE_BETWEEN_LOCAL_TIME_AND_UTC;
 const validateEntity = process.env.VALIDATE_ENTITIES;
 const userService = require(ROOT_PATH + '/generics/services/users');
-const userRolesHelper = require(MODULES_BASE_PATH + '/userRoles/helper');
-const programUsersHelper = require(MODULES_BASE_PATH + '/programUsers/helper');
```
Comment (VISHNUDAS-tunerlabs) [inline module/programs/helper.js:14]: @praveenKDass please confirm that these imports are not required

**Review by VISHNUDAS-tunerlabs — CHANGES_REQUESTED** [review]
Reviewed on 11 sept - 2024 3:05 PM

File: helper/pdfGeneration.js (hunk 3 above)
Comment (praveenKDass) [reply helper/pdfGeneration.js:1152]: removed console.log @VISHNUDAS-tunerlabs

File: module/programs/helper.js (hunk 5 above)
Comment (praveenKDass) [reply module/programs/helper.js:14]: not Required @VISHNUDAS-tunerlabs

**Review by VISHNUDAS-tunerlabs — APPROVED** [review]
Reviewed 11-sept 6:00 PM

**Pull request merged**